ONSdigital/rm-sample-service
# adding dead letter subscriber

# What and why?
Adding a new subscriber to deal with samples added to the dead letter queue after 10 failed processing attempts in `csv-worker`. This will allow the sample to be set to the `FAILED` state and the user will be alerted to an issue having happened rather than only being told to refresh as the sample would be stuck in the `INIT` state.
# How to test?
- Deploy this PR (using `adding_sample_file_dead_letter_sub` as the config branch) and the `csv-worker` PR on the trello card
- Download the broken sample file linked in the trello card
- Create a collection exercise and upload the sample file
- An error message should replace the loading text telling the user there is an issue with the file formatting

Note: for this to work the concourse pubsub job needs to be ran again while the terraform branch is set to `https://github.com/ONSdigital/ras-rm-terraform/pull/221`
# Trello
https://trello.com/c/bImN6iK4/1547-bug-problems-with-loading-a-sample
